nestkernel/structural_plasticity_node.h

```diff
@@ -0,0 +1,184 @@
+#include "nest_time.h"
+#include "nest_types.h"
+#include "node.h"
+#include "synaptic_element.h"
+
+// Includes from sli:
+#include "dictdatum.h"
+
+#define DEBUG_ARCHIVER 1
+
+namespace nest
+{
+
+/**
+ * \class Structural_Plasticity_Node
+ * a node which archives spike history for the purposes of
+ * timing dependent plasticity
+ */
+class Structural_Plasticity_Node : public Node
+{
+  using Node::get_synaptic_elements;
+
+public:
+  /**
+   * \fn Structural_Plasticity_Node()
+   * Constructor.
+   */
+  Structural_Plasticity_Node();
+
+  /**
+   * \fn Structural_Plasticity_Node()
+   * Copy Constructor.
+   */
+  Structural_Plasticity_Node( const Structural_Plasticity_Node& );
+  /**
+
+   * \fn double get_Ca_minus()
+   * return the current value of Ca_minus
+   */
+  double get_Ca_minus() const;
+
+  /**
+   * \fn double get_synaptic_elements(Name n)
+   * get the number of synaptic element for the current Node
+   * the number of synaptic elements is a double value but the number of
+   * actual vacant and connected elements is an integer truncated from this
+   * value
+   */
+  double get_synaptic_elements( Name n ) const;
+
+  /**
+   * \fn int get_synaptic_elements_vacant(Name n)
+   * Get the number of synaptic elements of type n which are available
+   * for new synapse creation
+   */
+  int get_synaptic_elements_vacant( Name n ) const;
+
+  /**
+   * \fn int get_synaptic_elements_connected(Name n)
+   * get the number of synaptic element of type n which are currently
+   * connected
+   */
+  int get_synaptic_elements_connected( Name n ) const;
+
+  /**
+   * \fn std::map<Name, double> get_synaptic_elements()
+   * get the number of all synaptic elements for the current Node
+   */
+  std::map< Name, double > get_synaptic_elements() const;
+
+  /**
+   * \fn void update_synaptic_elements()
+   * Change the number of synaptic elements in the node depending on the
+   * dynamics described by the corresponding growth curve
+   */
+  void update_synaptic_elements( double t );
+
+  /**
+   * \fn void decay_synaptic_elements_vacant()
+   * Delete a certain portion of the vacant synaptic elements which are not
+   * in use
+   */
+  void decay_synaptic_elements_vacant();
+
+  /**
+   * \fn void connect_synaptic_element()
+   * Change the number of connected synaptic elements by n
+   */
+  void connect_synaptic_element( Name name, int n );
+
+  void get_status( DictionaryDatum& d ) const;
+  void set_status( const DictionaryDatum& d );
+
+  /**
+   * retrieve the current value of tau_Ca which defines the exponential decay
+   * constant of the intracellular calcium concentration
+   */
+  double get_tau_Ca() const;
+
+protected:
+  /**
+   * \fn void set_spiketime(Time const & t_sp, double offset)
+   * record spike history
+   */
+  void set_spiketime( Time const& t_sp, double offset = 0.0 );
+
+  /**
+   * \fn void clear_history()
+   * clear spike history
+   */
+  void clear_history();
+
+private:
+  double last_spike_;
```

Comment:
Looks to me as if we have `last_spike_` now in `StructuralPlasticityNode` *and*  in `Archiving_Node`. Please make sure that it appears only one, presumably `Archiving_Node`.